Update package dependencies

rohitkrai03 opened the pull request:
## Fixes
- https://issues.redhat.com/browse/HAC-4882

## This PR -
- Updates the dependencies for hac-core.

florkbr commented on frontend/src/Utils/Loader.tsx:
```diff
@@ -1,5 +1,6 @@
 import * as React from 'react';
-import { Bullseye, Spinner } from '@patternfly/react-core';
+import { Bullseye } from '@patternfly/react-core/dist/dynamic/layouts/Bullseye';
```
@rohitkrai03 I believe we delivered a change that does this for us already at build time: https://github.com/patternfly/patternfly-react/pull/9298 - though we should confirm.

rohitkrai03 replied:
I had to make this change because there were lint errors without this.

rohitkrai03 replied:
I think this is coming from `forbid-pf-relative-imports` rule in `@redhat-cloud-services/eslint-config-redhat-cloud-services`.

florkbr commented on frontend/src/Routes/DynamicRoute/DynamicRoute.tsx:
```diff
@@ -2,8 +2,10 @@ import * as React from 'react';
 import { isRoutePage as isDynamicRoutePage, RoutePage as DynamicRoutePage, isContextProvider } from '@openshift/dynamic-plugin-sdk-extensions';
-import { Bullseye, Spinner, PageSection } from '@patternfly/react-core';
-import { ErrorState } from '@redhat-cloud-services/frontend-components/ErrorState';
+import { Bullseye } from '@patternfly/react-core/dist/dynamic/layouts/Bullseye';
```
Again I'd check out https://github.com/patternfly/patternfly-react/pull/9298

rohitkrai03 replied:
I think this is coming from forbid-pf-relative-imports rule in @redhat-cloud-services/eslint-config-redhat-cloud-services.

florkbr commented on frontend/package.json:
```diff
@@ -26,17 +26,17 @@
-    "@redhat-cloud-services/frontend-components": "^3.8.10",
-    "@redhat-cloud-services/frontend-components-notifications": "^3.2.14",
-    "@redhat-cloud-services/frontend-components-utilities": "^3.2.14",
+    "@patternfly/patternfly": "^5.0.2",
```
Can we bump these to latest:
```
"@patternfly/patternfly": "^5.0.4",
"@patternfly/react-core": "^5.0.1",
"@patternfly/react-table": "^5.0.1",
```
Not required as we have the carat, but if we're pushing another commit it would be nice

nicolethoen replied:
For what it's worth @florkbr and @rohitkrai03,
We did just have a PF release last week:

It addresses some UI bugs and introduces a few new features ([release notes](https://www.patternfly.org/get-started/release-highlights#patternfly-5.1))
 
"@patternfly/patternfly": "5.1.0",
"@patternfly/react-charts": "7.1.1",
"@patternfly/react-code-editor": "5.1.1",
"@patternfly/react-core": "5.1.1",
"@patternfly/react-icons": "5.1.1",
"@patternfly/react-styles": "5.1.1",
"@patternfly/react-table": "5.1.1",
"@patternfly/react-tokens": "5.1.1".
"@patternfly/react-topology": "5.1.0"

rohitkrai03 replied:
Updated to latest.

florkbr commented on frontend/package.json:
```diff
@@ -26,17 +26,17 @@
   },
   "dependencies": {
-    "@openshift/dynamic-plugin-sdk": "^2.0.1",
+    "@openshift/dynamic-plugin-sdk": "^3.0.0",
```
We're also on `"@openshift/dynamic-plugin-sdk": "^4.0.0"` now.

rohitkrai03 replied:
The chrome and scalprum packages do not use the latest SDK yet. That's why using v3.

rohitkrai03 replied:
Updated to latest.

florkbr commented on frontend/src/store/index.ts:
```diff
@@ -12,19 +13,19 @@ export type Registry = {
     undefined,
   );
-  registry.register(SDKReducers);
+  registry.register(SDKReducers as any);
```
I'm curious as to why we need to cast to any here. Do we need a type change in the SDK?

rohitkrai03 replied:
There was a type mismatch between what SDK was providing vs what `ReducerRegistry` expected.

rohitkrai03 commented:
@florkbr @karelhala I've updated the PR with latest versions and removed WIP. But we need to merge this along with the HAC Dev dependencies update PR because there is a breaking change in `dynamic-plugin-sdk-utils` package.

karthikjeeyar commented:
/lgtm
